pkg/recipes/terraform/execute.go

```diff
@@ -281,7 +281,7 @@ func downloadAndInspect(ctx context.Context, tf *tfexec.Terraform, options Optio
 	// Load the downloaded module to retrieve providers and variables required by the module.
 	// This is needed to add the appropriate providers config and populate the value of recipe context variable.
 	logger.Info(fmt.Sprintf("Inspecting the downloaded Terraform module: %s", options.EnvRecipe.TemplatePath))
-	loadedModule, err := inspectModule(tf.WorkingDir(), options.EnvRecipe.Name)
+	loadedModule, err := inspectModule(tf.WorkingDir(), options.EnvRecipe.Name, options.EnvRecipe.TemplatePath)
```

Comment:
Just thinking out loud; would it make more sense to have the whole Recipe object as the input to the **inspectModule** function? For future proofing the function... Instead of increasing the number of parameters, we can have a single parameter that would contain all the necessary.

Reply:
Updated to use the recipe